build.sbt

```diff
@@ -11,11 +11,27 @@ val sbt10ScalaVersion     = "2.12.4"
 val libScalaVersion       = "2.11.12"
 val libCrossScalaVersions = Seq("2.11.8", "2.11.11", libScalaVersion)
 
-lazy val baseSettings = Seq(
-  organization := "org.scala-native",
-  version := nativeVersion
+// Convert "SomeName" to "some-name".
+def convertCamelKebab(name: String) : String = {
+  name.replaceAll( "([a-z])([A-Z]+)", "$1-$2" ).toLowerCase
+}
+
+// Generate project name from project id.
+def projectName(project: sbt.ResolvedProject) : String = {
+  convertCamelKebab(project.id)
+}
+
+// Provide consistent project name pattern.
+lazy val nameSettings = Seq(
```

Comment:
This seems to belong to `publishSettings`. Lets put it there instead, then you don't have to add extra `.settings(nameSettings)` everywhere.

Reply:
applied to both `publishSettings` and 'noPublishSettings' since there is no shared settings root in this `build.sbt`, which could cover all project modules

Reply:
👍